lib/date_time/date_time.c

```diff
@@ -9,14 +9,17 @@
 #include <zephyr/types.h>
 #include <device.h>
 #include <stdio.h>
-#include <stdlib.h>
+
 #if defined(CONFIG_DATE_TIME_MODEM)
 #include <nrf_modem_at.h>
+#if defined(CONFIG_DATE_TIME_AUTO_UPDATE)
+#include <modem/at_monitor.h>
+#endif
 #endif
 #if defined(CONFIG_LTE_LINK_CONTROL)
 #include <modem/lte_lc.h>
 #endif
-#include <time.h>
+
 #include <errno.h>
 #include <string.h>
 #include <net/sntp.h>
```

Comment:
I've had to flag SNTP includes with `CONFIG_DATE_TIME_NTP` due to type collisions in `native_posix` compilation that doesn't have NTP enabled.